appinventor/blocklyeditor/src/blocklyeditor.js

```diff
@@ -40,6 +40,32 @@ Blockly.configForTypeBlock = {
 Blockly.BlocklyEditor.render = function() {
 };
 
+function unboundVariableHandler(myBlock, yailText) {
+  var unbound_vars = Blockly.LexicalVariable.freeVariables(myBlock);
+  unbound_vars = unbound_vars.toList();
+  if (unbound_vars.length == 0) {
+    Blockly.ReplMgr.putYail(yailText, myBlock);
+  } else {
+    var form = "<form>" + Blockly.Msg.DIALOG_ENTER_VALUES + "<br>";
+    for (var v in unbound_vars) {
+      form  +=  unbound_vars[v] + ' = <input type=text name=' + unbound_vars[v] + '><select> <option value=string>' + Blockly.Msg.TYPE_STRING +
+        '</option><option value=integer>' + Blockly.Msg.TYPE_INT + '</option><option value=bool>' + Blockly.Msg.TYPE_BOOL + '</option><br>';
```

Comment:
In App inventor, the types are string, number, and boolean, respectively. Also, while you add the dropdown it doesn't look like you consume the information when doing the codegen.